src/metabase/lib/schema/expression/arithmetic.cljc

```diff
@@ -33,28 +34,94 @@
 (mr/def ::args.numbers
   [:repeat {:min 2} [:schema [:ref ::expression/number]]])
 
-(defn- plus-minus-schema [tag]
-  [:or
-   [:and
-    [:cat
-     [:= tag]
-     [:schema [:ref ::common/options]]
-     [:schema [:ref ::expression/temporal]]
-     [:repeat {:min 1} [:schema [:ref :mbql.clause/interval]]]]
-    [:fn
-     {:error/message "Temporal arithmetic expression with valid interval units for the expression type"}
-     (fn [[_tag _opts expr & intervals]]
-       (let [expr-type (expression/type-of expr)]
-         (every? #(valid-interval-for-type? % expr-type) intervals)))]]
+(defn- validate-plus-minus-temporal-arithmetic-expression
```

Comment:
Generate a descriptive error message if the clause has invalid args; see the `temporal-arithmetic-schema-test` test below for an example of the output.